tsconfig.build.json

```diff
@@ -0,0 +1,19 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
+    "alwaysStrict": true,
+    "declaration": true,
+    "esModuleInterop": true,
+    "resolveJsonModule": true,
+    "forceConsistentCasingInFileNames": true,
+    "lib": ["esnext"],
+    "module": "CommonJS",
+    "moduleResolution": "node",
+    "strict": true,
+    "target": "es6",
+    "rootDirs": ["src", "typechain-types", "abi/**/*.json"],
+    "outDir": "build"
+  },
+  "include": ["src", "typechain-types", "abi/**/*.json"],
+}
```

Comment:
we might want to talk about any of these that might not be necessary - if you want to play around with the build - you can pull down the repo and do `yarn build` to see what exports to the `build` folder

Reply:
Hmm. `yarn build` is failing for me:

```
$ tsc -p tsconfig.build.json
tasks/estimate_minting_gas.ts:133:46 - error TS2339: Property 'zkSyncDeploy' does not exist on type 'HardhatConfig'.

133     const provider = new Provider(hre.config.zkSyncDeploy.zkSyncNetwork);
                                                 ~~~~~~~~~~~~


Found 1 error.
```

Reply:
ah i updated this to not extend the tsconfig and to build just these folders (excluding hardhat.config.ts) - not sure if thats something we would want to export